### DISCUSSION TITLE
Is sampler1D designed to read the previous value?

### DISCUSSION THREAD

**Discussion by HSDSZ** [bnpr/Malt#255] [Q&A] [answered]
Even if I don't assign any value to sampler1D, it will still use the previous sampler1D value.

For example:
step1. Connect the diffuse gradient to the main output. Do not assign any value to the sampler1D. The color is black which is what it supposed to be if no input.
![image](https://user-images.githubusercontent.com/36077492/154917674-0b9b0dd1-056f-4a9b-99cd-fa822a8b8c77.png)

step2, If I assign any value to the sampler1D
![image](https://user-images.githubusercontent.com/36077492/154918209-5661d432-09a8-46b8-aa20-da1129acd8d0.png)

step3, then I remove the sampler1D. I suppose the color of the ball should return back to black. But actually, it preserve the previous diffuse color. 
![image](https://user-images.githubusercontent.com/36077492/154918328-1db9c324-a0e4-4e12-bc99-8f25678cbdf6.png)

**Comment by pragma37** [comment, marked as answer]
Nope! This was a bug.
It should be solved by 63f223996b44641effe93907901ebba389436e22.

**Reply by HSDSZ** [reply]
After this fix. The new development version is not usable. It keeps loading the MALT window. I will create a new issue
